src/task/src/runners/runnerInvoker.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { GitWritableStream } from '../git/gitWritableStream'
+import { singleton } from 'tsyringe'
+import AzurePipelinesRunnerInvoker from './azurePipelinesRunnerInvoker'
+import GitHubRunnerInvoker from './gitHubRunnerInvoker'
+import IRunnerInvoker from './iRunnerInvoker'
+
+/**
+ * A wrapper around the runner functionality, to facilitate testability. This class cannot use logging functionality as
+ * the logger forms part of the runner functionality, and using logging here could result in circular dependencies.
+ */
+@singleton()
+export default class RunnerInvoker implements IRunnerInvoker {
+  private readonly _azurePipelinesRunnerInvoker: AzurePipelinesRunnerInvoker
+  private readonly _gitHubRunnerInvoker: GitHubRunnerInvoker
+
+  private _runnerInvoker: IRunnerInvoker | undefined
+  private _localizationInitialized: boolean = false
+
+  /**
+   * Initializes a new instance of the `RunnerInvoker` class.
+   * @param azurePipelinesRunnerInvoker The Azure Pipelines runner logic.
+   * @param gitHubRunnerInvoker The GitHub runner logic.
+   */
+  public constructor (azurePipelinesRunnerInvoker: AzurePipelinesRunnerInvoker, gitHubRunnerInvoker: GitHubRunnerInvoker) {
+    this._azurePipelinesRunnerInvoker = azurePipelinesRunnerInvoker
+    this._gitHubRunnerInvoker = gitHubRunnerInvoker
+  }
+
+  /**
+   * Gets a value indicating whether a GitHub runner is in use.
+   */
+  public static get isGitHub (): boolean {
+    return process.env.GITHUB_ACTION !== undefined
```

Comment:
I think this answers my earlier question regarding GH vs. GHE behaviour.

Reply:
From reading the documentation, my understanding is that this will be present for all types of GitHub. It would be great to verify this, but I'm not sure how we can right now. If you have any ideas, I'd love to hear them.